secret/secret.go

```diff
@@ -91,18 +91,42 @@ func New(config *Config) (*Service, error) {
 	}, nil
 }
 
-// Seal takes plaintext and returns encrypted and authenticated ciphertext.
-func (s *Service) Seal(value []byte) (*SealedBytes, error) {
-	return s.SealWithKey(value, s.secretKey)
+// Seal takes plaintext and a key and returns encrypted and authenticated ciphertext.
+// Allows passing in a key and useful for one off sealing purposes, otherwise
+// create a secret.Service to seal multiple times.
+func Seal(value []byte, secretKey *[SecretKeyLength]byte) (*SealedBytes, error) {
+	// check that we either initialized with a key or one was passed in
```

Comment:
The "initialized with a key" part isn't relevant for the package-level function.